projects/igniteui-angular/src/lib/grids/grid/grid-validation.service.ts

```diff
@@ -32,6 +32,7 @@ export class IgxGridValidationService {
                 const value = resolveNestedPath(data || {}, col.field);
                 const field = this.getFieldKey(col.field);
                 const control = new FormControl(value, { updateOn: this.grid.validationTrigger });
+                control.setValue(value);
```

Comment:
Um.. if the control was just initialized with the same value, I'm kind of missing the point of assigning it again. Any specific reason?

Reply:
Yes. It's due to a rather weird behavior when the initial value is undefined (which we use when creating an adding row for example, all fields have undefined as value).  When the form control is created for undefined it actually sets it to null instead:

So this:
```
new FormControl(undefined, { updateOn: this.grid.validationTrigger });
```

Then returns null for control.value.

Which breaks some tests.